### PR TITLE
Context drawer

### PR DESCRIPTION
**Pull request by chri2547**

### CODE REVIEW

**Comment by cyrichardson** [comment]
On it, @chri2547.

**Comment by cyrichardson** [comment]
@chri2547 Suggestions here: 
[helix_context-drawer-cyrEdits.docx](https://github.com/rackerlabs/design-system/files/2877051/helix_context-drawer-cyrEdits.docx)

File: _component_design/context-drawer.md [hunk 1]
```diff
@@ -0,0 +1,539 @@
...
+
+The drawer slides out twice as fast as it slides in because the closing animation requires less user focus than does the user’s next task.
+
+**Animating into view**
+
+To calculate animation duration for sliding the drawer into view, use the following formula:
+
+*SlideInDuration = (DrawerWidthPX * 0.8)/1000*
+
+For example: STILL NEED AN EXAMPLE FROM DREW
```
Comment (araiford) [inline _component_design/context-drawer.md]: Something along these lines:

The medium width drawer is 512px wide at the extra-large breakpoint, so the formula is as follows:
SlideInDuration = (512 * 0.8)/1000 = 0.4096 seconds

File: _component_design/context-drawer.md [hunk 2]
```diff
@@ -0,0 +1,539 @@
...
+
+**Animating out of view**
+
+The animation duration to slide out is calculated as follows:
+
+*SlideOutDuration = SlideInDuration/2*
+
+For example: STILL NEED AN EXAMPLE FROM DREW
```
Comment (araiford) [inline _component_design/context-drawer.md]: The medium width drawer is 512px wide at the extra-large breakpoint, so the formula is as follows:
SlideOutDuration = ((512 * 0.8)/1000)/2 = 0.2048 seconds

File: _component_design/context-drawer.md [hunk 3]
```diff
@@ -0,0 +1,539 @@
...
+Hide the footer until the contents loads.
+
+When the content has loaded, hide the spinner and show the footer if there is one.
+
+{% endcolumn %}
+
+{% column right:"hxCol hxSpan-12-xs hxSpan-12-sm hxSpan-8-md hxSpan-8-lg" %}
+
+{% figure [caption:"Context drawer loading state"] [class:"image bg-light border"] %}
+<embed src="{{site.baseurl}}/assets/images/components/content-areas/context-drawer/context-drawer-states-loading.png" width="436"/>
```
Comment (araiford) [inline _component_design/context-drawer.md]: I noticed a few small inconsistencies in the drawer images.  I'm going to send you new images for a few of these.

Once that's done, I think we're good to go.

File: _component_design/context-drawer.md (hunk 1 above)
Comment (chri2547) [reply _component_design/context-drawer.md]: Done.

File: _component_design/context-drawer.md (hunk 2 above)
Comment (chri2547) [reply _component_design/context-drawer.md]: Done.

File: _component_design/context-drawer.md (hunk 3 above)
Comment (chri2547) [reply _component_design/context-drawer.md]: Cool, just let me know when those are available!

Comment (chri2547) [reply _component_design/context-drawer.md]: visuals replaced, done.

**Comment by araiford** [comment]
:shipit:

File: _component_design/context-drawer.md [hunk 4]
```diff
@@ -0,0 +1,543 @@
...
+
+#### Duration
+
+Duration is the amount of time the drawer takes to animate from off-stage to on-stage, and from on-stage to off-stage.
+
+Each drawer size has a unique animation duration so that the visual weight and movement looks natural.
+
+The drawer slides out twice as fast as it slides in because the closing animation requires less user focus than does the user’s next task.
+
+**Animating into view**
```
Comment (nimbinatus) [inline _component_design/context-drawer.md]: Any reason for not using heading 5 (so people can link directly to these ones)?

File: _component_design/context-drawer.md [hunk 5]
```diff
@@ -0,0 +1,543 @@
...
+
+To calculate animation duration for sliding the drawer into view, use the following formula:
+
+*SlideInDuration = (DrawerWidthPX * 0.8)/1000*
+
+For example, at the extra-large breakpoint, the medium width drawer is 512px wide. The slide in duration is calculated as:
+
+*(512 * 0.8)/1000 = 0.4096 seconds*
+
+**Animating out of view**
```
Comment (nimbinatus) [inline _component_design/context-drawer.md]: Same question.

File: _component_design/context-drawer.md (hunk 4 above)
Comment (chri2547) [reply _component_design/context-drawer.md]: Done.

File: _component_design/context-drawer.md (hunk 5 above)
Comment (chri2547) [reply _component_design/context-drawer.md]: Done.

**Pull request merged**